src/app/pages/component-sidenav/component-sidenav.service.ts

```diff
@@ -0,0 +1,13 @@
+import {Injectable} from '@angular/core';
+import {MatSidenav} from '@angular/material/sidenav';
+
+@Injectable({providedIn : 'root'})
+export class ComponentSidenavService {
+  private _sidenav: MatSidenav;
+
+  constructor() {}
+
+  get sidenav(): MatSidenav { return this._sidenav; }
+
+  set sidenav(value: MatSidenav) { this._sidenav = value; }
```

Comment:
We try to only use getters/setters for `@Input` properties. This one looks like it could just be a regular property.